packages/block-library/src/heading/heading-level-warning.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * WordPress dependencies
+ */
+import { speak } from '@wordpress/a11y';
+import { Notice } from '@wordpress/components';
+import { useEffect } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+const INVALID_LEVEL_MESSAGE = __(
+	'The chosen heading level may be invalid. See the content structure tool for more info.'
+);
+
+const INVALID_LEVEL_MESSAGE_SPOKEN = __(
+	'The chosen heading level may be invalid. Use proper heading levels to organize your content for visitors and search engines. See the content structure tool for more info.'
+);
+
+export default function HeadingLevelWarning( {
+	selectedLevel,
```

Comment:
If the `selectedLevel` property will not be used eventually in the message, maybe we can remove it from the component.

Reply:
It's still being used to detect when the selected level has changed, so it can trigger a replay of the warning for screen reader users. Because of that, I don't think I can remove it as a dependency unless I extract the screen reader message out of the component. I think they should remain in the same component, however, since the message is essentially just the spoken version of the visual text.